### PR TITLE
feat: adds OverflowScroll, useOverflowScroll, CardCarousel

### PR DESCRIPTION
**Pull request by adamstankiewicz**
## Description

Adds the following:
* Documentation page: `useIsVisible`
* React hook (with documentation): `useOverflowScroll`
* React hook: `useOverflowScrollItems`
* React component (with documentation): `OverflowScroll`
  * React context: `OverflowScrollContext`
  * React component: `OverflowScroll.StartSentinel`
  * React component: `OverflowScroll.EndSentinel`
  * React component: `OverflowScroll.Items`
* React component: `CardCarousel` 

Updates the following:
* `CardDeck`
* Shows example on documentation of using `OverflowScroll` to build a `Chip` carousel (in the future, there might be a standalone `ChipCarousel` component, but this is a start).

**Note: tests are still a WIP.**

### Testing Instructions
1. Checkout the `ags/overflowscroll-cardcarousel` branch locally.
2. Run `npm i` to install top-level dependencies.
3. Run `cd www && npm i` to install dependencies for the documentation website.
4. Run `npm start` to start the documentation website.
5. Navigate to and verify the following:
  * http://localhost:8000/components/overflowscroll/overflowscroll/
  * http://localhost:8000/components/overflowscroll/useoverflowscroll/
  * http://localhost:8000/components/card/#carddeck
  * http://localhost:8000/components/card/#cardcarousel
  * http://localhost:8000/components/chip/#chipcarousel

### Deploy Preview

https://deploy-preview-1884--paragon-openedx.netlify.app/

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: src/Card/CardCarousel/CardCarousel.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import CardCarouselHeader from './CardCarouselHeader';
+import CardCarouselItems from './CardCarouselItems';
+
+function CardCarousel({
+  children,
+  title,
+  subtitle,
+  columnSizes,
+  hasInteractiveChildren,
+  canScrollHorizontal,
```
Comment (brobro10000) [inline src/Card/CardCarousel/CardCarousel.jsx:15]: Just some initial testing locally. I noticed a small ui issue when horizontal scroll is enabled, but the number of elements in the OverflowScroll is does not require horizontal scrolling. The container for the horizontal scroll is remaining despite not being able to scroll through the card carousel

Screen Shot 2023-01-03 at 9 50 49 AM

File: src/OverflowScroll/useOverflowScroll.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+* `childQuerySelector`. A CSS query selector that can be used to find the child elements within the overflow container.
+* `disableScroll`. An optional flag to disable the scroll behavior for the overflow container.
+* `disableOpacityMasks`. An optional flag to disable the opacity masks at the start/end of the overflow container (if applicable).
+
+The hook returns the following:
+* Refs
+  * `overflowRef`. The ref to pass to your overflow container. Note that this element may receive style overrides for `position: relative;` and `overflow-x: scroll` (optionally).
+  * `startSentinelRef`. The ref for an element representing the start sentinel of the overflow container. Used to determine whether the overflow container is scrolled all the way to the start.
+  * `endSentinelRef`. The ref for an element representing the end sentinel of the overflow container. Used to determine whether the overflow container is scrolled all the way to the end.
+* Scoll state
```
Comment (brobro10000) [inline src/OverflowScroll/useOverflowScroll.mdx:25]: Small typo

File: src/Card/CardCarousel/CardCarousel.jsx (hunk 1 above)
Comment (adamstankiewicz) [reply src/Card/CardCarousel/CardCarousel.jsx:15]: Going to defer on this for now.

**Review by brobro10000 — APPROVED** [review]
Nicely done, LGTM!

**Review by brobro10000 — APPROVED** [review]

**Pull request merged**